product/domain/productBasics.go

```diff
@@ -85,11 +85,12 @@ type (
 
 	// Saleable are properties required for being selled
 	Saleable struct {
-		IsSaleable      bool
-		SaleableFrom    time.Time
-		SaleableTo      time.Time
-		ActivePrice     PriceInfo
-		AvailablePrices []PriceInfo
+		IsSaleable         bool
+		SaleableFrom       time.Time
+		SaleableTo         time.Time
+		ActivePrice        PriceInfo
+		AvailablePrices    []PriceInfo
+		ActiveLoyaltyPrice LoyaltyPriceInfo
```

Comment:
```suggestion
		ActiveLoyaltyPrice *LoyaltyPriceInfo
```

Shouldnt this be a pointer for products that don't have a loyalty price?